recipes/librtlsdr/all/conanfile.py

```diff
@@ -0,0 +1,130 @@
+        extracted_dir = glob.glob(self.name + "-*/")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(
+            args=["-DCMAKE_BUILD_TYPE={}".format(self.settings.build_type), "-DCMAKE_VERBOSE_MAKEFILE=ON"],
+            source_folder=self._source_subfolder,
+            build_folder=self._build_subfolder,
+        )
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        os.rename("Findlibusb.cmake", "Findlibusb_.cmake")
+        cmake = self._configure_cmake()
+        # Do not build tools
+        cmake.build(target="rtlsdr_shared" if self.options.shared else "rtlsdr_static")
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        self.copy(
+            "*.h", dst="include", src=os.path.join(self._source_subfolder, "include")
+        )
+        self.copy("*.lib", src=self._build_subfolder, dst="lib", keep_path=False)
+        self.copy("*.dll", src=self._build_subfolder, dst="bin", keep_path=False)
+        self.copy(
+            "*.so*",
+            src=self._build_subfolder,
+            dst="lib",
+            keep_path=False,
+            symlinks=True,
+        )
+        self.copy("*.dylib", src=self._build_subfolder, dst="lib", keep_path=False)
+        self.copy("*.a", src=self._build_subfolder, dst="lib", keep_path=False)
+        for name in ("shared", "static"):
+            for extension in (".lib", ".so", ".dylib", ".a"):
+                try:
+                    tools.rename(
+                        os.path.join(
+                            self.package_folder,
+                            "lib/rtlsdr_{}{}".format(name, extension),
+                        ),
+                        os.path.join(
+                            self.package_folder, "lib/librtlsdr{}".format(extension)
+                        ),
+                    )
+                except:
+                    pass
+            try:
+                tools.rename(
+                    os.path.join(self.package_folder, "bin/rtlsdr_{}.dll".format(name)),
+                    os.path.join(self.package_folder, "bin/librtlsdr.dll"),
+                )
+            except:
+                pass
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        if self.settings.os in ("Linux", "FreeBSD", "SunOS"):
+            self.cpp_info.system_libs.extend(["pthread", "m"])
+        if self.options.shared:
+            self.cpp_info.defines.extend(["rtlsdr_SHARED=1"])
+        else:
+            self.cpp_info.defines.extend(["rtlsdr_STATIC=1"])
+        # Necessary to libusb
+        if self.settings.os in ["Linux", "Android"]:
+            self.cpp_info.system_libs.append("udev")
```

Comment:
Isn't `udev` on Linux a system package? systemd?

Reply:
Libusb's recipe has those lines. I added them too here becase the *very poor* way this recipe was build, I had to patch around to make it find conan's libusb and transitive dependencies aren't being handled properly.

Reply:
Without it it would fail to link with `udev_*` functions...

Reply:
Shouldn't librtlsdr inherit the flags of libusb and thus add udev automatically?
I don't think librtlsdr calls `udev_xxx` functions?

Reply:
it seems like `CMakeLists.txt` has been patched in this recipe, that caused CMake to fail to propagate some linker flags from the transitive target requirement. usually, CMake should be able to handle that automatically, and explicit flags shouldn't be required.

Reply:
The problem is in upstreams cmake/Modules/findLibUSB.cmake. It's a very bad piece of cmake.... I tried patching it out, whilst keeping the original functionality (like those two function checks)... It's just... Bad.

Reply:
I think I fixed it, let's see.